include/aikido/distance/ConfigurationRanker.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef AIKIDO_DISTANCE_CONFIGURATIONRANKER_HPP_
+#define AIKIDO_DISTANCE_CONFIGURATIONRANKER_HPP_
+
+#include <dart/dynamics/dynamics.hpp>
+#include "aikido/statespace/ScopedState.hpp" // TODO: Remove this.
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+namespace aikido {
+namespace distance {
+
+class ConfigurationRanker
+{
+public:
+  /// Constructor
+  ///
+  /// \param[in] metaSkeletonStateSpace Statespace of the skeleton.
+  /// \param[in] metaskeleton Metaskeleton of the robot.
+  /// \param[in] ikSolutions List of IK solutions to be ranked.
+  ConfigurationRanker(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
+      const std::vector<statespace::StateSpace::State*> ikSolutions);
+
+  /// Destructor
+  virtual ~ConfigurationRanker() = default;
+
+  /// Returns the statespace.
+  statespace::ConstStateSpacePtr getStateSpace() const;
+
+  /// Returns the vector of ranked IK solutions.
+  std::vector<std::pair<statespace::StateSpace::State*, double>>&
```

Comment:
I think this method might be more useful if it's something like `std::vector<State*>& rank(std::vector<State*>& ikSolutions)`. Perhaps it can take in the vector of states and sort them in place? I don't think it's necessary for an external user to see the resulting scores.